tslearn/nonmyopic.py

```diff
@@ -0,0 +1,370 @@
+        Maximum number of iterations performed by the MLP
+
+    minimum_time_stamp : int
+        Earliest time at which a classification can be performed on a time series
+
+    lamb : float
+        Value of the hyper parameter lambda used during the computation of the cost function to evaluate the probability
+        that a time series belongs to a cluster given the time series.
+
+    cost_time_parameter : float
+        Parameter of the cost function of time. This function is of the form : f(time) = time * cost_time_parameter
+
+    Attributes
+    --------------------
+    silhouette_ : float
+        The silhouette score from the clustering
+
+    classifier_ : list
+        A list containing all the classifiers trained for the model
+
+    clusters_ : dictionary
+        Contains the times series by clusters
+
+    pyhatyck_ : dictionary
+        Contains the probabilities of being classified as class y_hat given class y and cluster ck
+
+    indice_ck_ : list
+        Contains for each clusters the indexes of the time series in the dataset
+
+    __n_classes_ : int private
+        The number of classes in the series
+
+    __len_X_ : int
+        The length of the time series
+
+    Examples
+    --------------------
+
+    References
+    --------------------
+    [1] A. Dachraoui, A. Bondu & A. Cornuéjols. Early classification of time series as a non myopic sequential decision
+    making problem. 2015 Conference paper
+
+
+    """
+
+    def __init__(
+        self,
+        n_clusters,
+        solver,
+        hidden_layer_sizes,
+        random_state,
+        maximum_iteration,
+        minimum_time_stamp,
+        lamb,
+        cost_time_parameter,
+        classifier
+    ):
+        super(NonMyopicEarlyClassification, self).__init__()
+        self.solver = solver
+        self.hidden_layer_sizes = hidden_layer_sizes
+        self.random_state = random_state
+        self.maximum_iteration = maximum_iteration
+        self.n_clusters = n_clusters
+        self.minimum_time_stamp = minimum_time_stamp
+        self.lamb = lamb
+        self.cost_time_parameter = cost_time_parameter
+        self.classifier = classifier
+
+    def fit(self, X, Y):
+        """
+        This function fits classifiers that are currently multilayer perceptrons to a training set of time series and
+        associated classes. A classifier is fit for each time stamp above a minimum time stamp that is an attribute of
+        the class of the model. Then some probabilities are computed using clusters already computed.
+        This function should be divided among 2 or 3 functions.
+
+        Parameters
+        ----------
+        X: Array-like
+            a dataset of time series
+        Y: vector
+            the associated classes of the series from X_train
+        """
+        classes_y = np.unique(Y)
+        self.cluster_ = TimeSeriesKMeans(n_clusters=self.n_clusters)
+        self.classifier_ = []
+        self.clusters_ = {}
```

Comment:
Using variable names that are too similar (`self.cluster_` and `self.clusters_`) should be avoided in order to avoid bugs.